sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsDiskStore.scala

```diff
@@ -36,31 +36,37 @@ class RapidsDiskStore(
   override def createBuffer(
       incoming: RapidsBuffer,
       stream: Cuda.Stream): RapidsBufferBase = {
-    val incomingBuffer = incoming.getMemoryBuffer
-    try {
-      val hostBuffer = incomingBuffer match {
-        case h: HostMemoryBuffer => h
-        case _ => throw new UnsupportedOperationException("buffer without host memory")
-      }
-      val id = incoming.id
-      val path = if (id.canShareDiskPaths) {
-        sharedBufferFiles.computeIfAbsent(id, _ => id.getDiskPath(diskBlockManager))
-      } else {
-        id.getDiskPath(diskBlockManager)
-      }
-      val fileOffset = if (id.canShareDiskPaths) {
-        // only one writer at a time for now when using shared files
-        path.synchronized {
-          copyBufferToPath(hostBuffer, path, append = true)
+    if (incoming.alreadySpilledToDisk) {
```

Comment:
Checking for a buffer already being spilled in the disk store seems backwards to me.   We'd have to put this type of check in every type of buffer store except the device store, since it's not guaranteed that the spill went from device to disk.  It could be in host memory or some other store we invent later.

IMO a cleaner design is to check for the already-been-spilled logic in the one place that realy needs to do it: the device store.  The device store can simply free the buffer if it's already been spilled to _any store_, rather than checking for this in every other store except the device store.

Reply:
I would be fine with that too.

Reply:
Done.